src/uwtools/templater.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+'''
+This utility renders a Jinja2 template using user-supplied configuration options
+via YAML or environment variables.
+'''
+
+import os
+import sys
+import argparse
+
+from jinja2 import Environment, FileSystemLoader, meta
+import f90nml
+
+from uwtools import config
+
+def file_exists(arg):
+    ''' Checks whether a file exists, and returns the path if it does. '''
+    if not os.path.exists(arg):
+        msg = f'{arg} does not exist!'
+        raise argparse.ArgumentTypeError(msg)
+    return arg
+
+def parse_args(argv):
+
+    '''
+    Function maintains the arguments accepted by this script. Please see
+    Python's argparse documentation for more information about settings of each
+    argument.
+    '''
+
+    parser = argparse.ArgumentParser(
+       description='Update a Jinja2 Template with user-defined settings.'
+    )
+    parser.add_argument(
+        '-o', '--outfile',
+        help='Full path to output file',
+        )
+    parser.add_argument(
+        '-i', '--input_template',
+        help='Path to a Jinja2 template file.',
+        required=True,
+        type=file_exists,
+        )
+    parser.add_argument(
+        '-c', '--config_file',
+        help='Optional path to a YAML configuration file. If not provided, '
+        'os.environ is used to configure.',
+        type=file_exists,
+        )
+    parser.add_argument(
+        '-d', '--dry_run',
+        action='store_true',
+        help='If provided, print rendered template to stdout only',
+        )
+    parser.add_argument(
+        '--values_needed',
+        action='store_true',
+        help='If provided, print a list of required configuration settings to stdout',
+        )
+    return parser.parse_args(argv)
+
+def set_template(argv):
+    '''Main section for set_namelist ingest utility'''
+
+    cla = parse_args(argv)
```

Comment:
It's not clear to me what `cla` stands for here.

Reply:
I've used it as another acronym for command line args. I am considering user_args instead. What do you think?

Reply:
Yep I think that works!